server/tests/helpers/core-utils.ts

```diff
@@ -0,0 +1,47 @@
+/* tslint:disable:no-unused-expression */
+
+import * as chai from 'chai'
+import 'mocha'
+import {
+  parseBytes
+} from '../../helpers/core-utils'
+
+const expect = chai.expect
+
+describe('Parse Bytes', function () {
+  it('Should pass when given valid value', async function () {
+    // just return it
+    expect(parseBytes(1024)).to.be.eq(1024)
+    expect(parseBytes(1048576)).to.be.eq(1048576)
+    expect(parseBytes('1024')).to.be.eq(1024)
+    expect(parseBytes('1048576')).to.be.eq(1048576)
+    
+    // sizes
+    expect(parseBytes('1MB')).to.be.eq(1024)
+    expect(parseBytes('1GB')).to.be.eq(1048576)
+    expect(parseBytes('1TB')).to.be.eq(1073741824)
+
+    expect(parseBytes('5GB')).to.be.eq(5242880)
+    expect(parseBytes('5TB')).to.be.eq(5368709120)
+
+    expect(parseBytes('1024MB')).to.be.eq(1048576)
+    expect(parseBytes('1024GB')).to.be.eq(1073741824)
+    expect(parseBytes('1024TB')).to.be.eq(1099511627776)
+    
+    // with whitespace
+    expect(parseBytes('1 GB')).to.be.eq(1048576)
+    expect(parseBytes('1\tGB')).to.be.eq(1048576)
+
+    // sum value
+    expect(parseBytes('1TB 1024MB')).to.be.eq(1074790400)
+    expect(parseBytes('4GB 1024MB')).to.be.eq(5242880)
+    expect(parseBytes('4TB 1024GB')).to.be.eq(5368709120)
+    expect(parseBytes('4TB 1024GB 0MB')).to.be.eq(5368709120)
+    expect(parseBytes('1024TB 1024GB 1024MB')).to.be.eq(1100586418176)
```

Comment:
I never saw this kind of format used anywhere. Is it really necessary to support?

Reply:
If you remove this functionality, you could use a simple switch for `parseBytes()`, and get rid of the complicated regex stuff.

Reply:
We are using integers all over the place, and in the config some tuff is commented out where  it uses the strings but they aren't parsed. So this functionality is pretty much needed the way things are headed

Reply:
```
redundancy:
  videos:
    check_interval: '1 hour' # How often you want to check new videos to cache
    strategies:
#      -
#        size: '10GB'
#        # Minimum time the video must remain in the cache. Only accept values > 10 hours (to not overload remote instances)
#        min_lifetime: '48 hours'
#        strategy: 'most-views' # Cache videos that have the most views
#      -
#        size: '10GB'
#        # Minimum time the video must remain in the cache. Only accept values > 10 hours (to not overload remote instances)
#        min_lifetime: '48 hours'
#        strategy: 'trending' # Cache trending videos
#      -
#        size: '10GB'
#        # Minimum time the video must remain in the cache. Only accept values > 10 hours (to not overload remote instances)
#        min_lifetime: '48 hours'
#        strategy: 'recently-added' # Cache recently added videos
#        min_views: 10 # Having at least x views
```

Reply:
also to specify the quotas, instead of using bytes be can just give it a string value

```
user:
  # Default value of maximum video BYTES the user can upload (does not take into account transcoded files).
  # -1 == unlimited
  video_quota: -1
  video_quota_daily: -1
```

Reply:
If you can come up with a better way to do it without regex, go for it

Reply:
I meant the part with having TB and GB both in the same value. If you leave that out, you can just do something like '10 GB'.split(' ')` and do a switch on the second part.